service/src/main/kotlin/fi/espoo/evaka/shared/security/AccessControl.kt

```diff
@@ -291,7 +294,10 @@ WHERE employee_id = :userId
                     ).read { it.wasUploadedByAnyEmployee(id) }
                 Action.Attachment.READ_MESSAGE_DRAFT_ATTACHMENT,
                 Action.Attachment.DELETE_MESSAGE_DRAFT_ATTACHMENT ->
-                    Database(jdbi).read { it.isOwnAttachment(id, user) }
+                    Database(jdbi).read { it.hasPermissionForAttachmentThroughMessageDraft(user, id) }
```

Comment:
`is AuthenticatedUser.Citizen -> Database(jdbi).read { it.isOwnAttachment(id, user) }`
Not sure if part of this PR, but this should also let the user read attachments in received messages

Reply:
I can continue with that in my PR